### PR TITLE
[DPC-3581] Set up Tealium for dpc.cms.gov

### PR DESCRIPTION
**Pull request by ashley-weaver**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-3581

## 🛠 Changes

(What was added, updated, or removed in this PR.)

## ℹ️ Context for reviewers

(Background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers.)

## ✅ Acceptance Validation

(How were the changes verified? Did you fully test the acceptance criteria in the ticket? Provide reproducible testing instructions and screenshots if applicable.)

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

**Review by jdettmannnava — COMMENTED** [review]
In the implemetation doc, there is a utag_layer element; is this unnecessary?
Also in the doc, the async script is in the body rather than the head, although I am unsure if this matters.

**Comment by ashley-weaver** [comment]
@jdettmannnava I followed how the setup was done for ab2d-website, as per @SravanKDuddella's guidance

File: _includes/_head.html [hunk 1]
```diff
@@ -14,4 +14,17 @@
...
   <link rel="stylesheet" href="{{ site.cssurl | relative_url }}" />
     <!-- Global site tag (gtag.js) - Google Analytics -->
+
+  <!-- Tealium Tag Manager - SYNC -->
+  <script src="https://tags.tiqcdn.com/utag/cmsgov/cms-dpc/prod/utag.sync.js"></script>
+
+  <!-- Tealium Tag Manager - ASYNC -->
+  <script type="text/javascript">
+    (function(a,b,c,d){
+    a='https://tags.tiqcdn.com/utag/cmsgov/cms-dpc/prod/utag.js';
```
Comment (kyeah) [inline _includes/_head.html:24]: I do think we want to set the `utag_data` global var / separate synchronous script before we load this, as mentioned in the confluence instructions and in the email from Josh. 

It looks like there's a utag_data already set up for BCDA, although it's still from the Google Analytics days so the actual `utag_data` keys are different.

https://github.com/CMSgov/bcda-static-site/blob/7795a67343880880f23faa374bfe3902b235bd25/_layouts/default.html#L8

For us it should look a bit like this:

```js
var utag_data = {
  "content_language": "en",
  "page_name": "{{ page.title }}",
  "page_path": "{{ page.url }}",
  "site_domain": "https://dpc.cms.gov",  // or "https://stage.dpc.cms.gov"
  "site_environment": "{{ jekyll.environment }}",
}
```

When I follow the instructions in the Confluence page to look in the console, there's a noticeable difference in the events with and without this set up.

|With|Without utag_data|
|---|---|
|Screen Shot 2023-10-11 at 5 57 57 PM|Screen Shot 2023-10-11 at 6 01 26 PM|

There are more events emitted after the pre-init step that all contain the `utag_data` when it's set.

Comment (ashley-weaver) [reply _includes/_head.html:24]: ah, thanks, I was a little confused about where the body template was

File: _includes/_head.html [hunk 2]
```diff
@@ -14,4 +14,17 @@
 
   <link rel="stylesheet" href="{{ site.cssurl | relative_url }}" />
     <!-- Global site tag (gtag.js) - Google Analytics -->
+
+  <!-- Tealium Tag Manager - SYNC -->
+  <script src="https://tags.tiqcdn.com/utag/cmsgov/cms-dpc/prod/utag.sync.js"></script>
```
[the remark this answers is not in the document]
Comment (ashley-weaver) [reply _includes/_head.html:19]: is there a way to parametrize the script source like that (ie "cms-dpc/{{env}}/utag.sync.js")?

[the remark this answers is not in the document]
Comment (ashley-weaver) [reply _includes/_head.html:19]: the error I'm getting now is `utag.sync.js:1     Failed to load resource: net::ERR_BLOCKED_BY_CLIENT`

[the remark this answers is not in the document]
Comment (kyeah) [reply _includes/_head.html:19]: > is there a way to parametrize the script source like that (ie "cms-dpc/{{env}}/utag.sync.js")?

yeah you should be able to use `{{ jekyll.environment }}`. You may need to map values using an if/else statement:

```suggestion
{% if jekyll.environment == 'prod' %}
  <script src="https://tags.tiqcdn.com/utag/cmsgov/cms-dpc/prod/utag.sync.js"></script>
{% elsif jekyll.environment == 'stage' %}
    <script src="https://tags.tiqcdn.com/utag/cmsgov/cms-dpc/qa/utag.sync.js"></script>
{% else %}
      <script src="https://tags.tiqcdn.com/utag/cmsgov/cms-dpc/dev/utag.sync.js"></script>
{% endif %}
```

> the error I'm getting now is utag.sync.js:1     Failed to load resource: net::ERR_BLOCKED_BY_CLIENT

what's the `src` that it's trying to load? wondering if the environment is not in the valid values ([prod, qa, dev])

[the remark this answers is not in the document]
Comment (ashley-weaver) [reply _includes/_head.html:19]: I'm actually not seeing the tealium script in the elements anymore, just getting that error on the console. I'll push up what I have.

[the remark this answers is not in the document]
Comment (kyeah) [reply _includes/_head.html]: This is what I'm seeing with the latest changes:

Screen Shot 2023-10-13 at 4 31 27 PM

Looks like I'm getting the script elements, but the `profile` isn't declared. I also see that we'll probably need similar `if / elsif / else` statements when checking jekyll.environment for the async script.

**Review by kyeah — APPROVED** [review]
Thank you! confirmed behavior locally.

Approving with a small update request.

File: _includes/_head.html [hunk 3]
```diff
@@ -4,6 +4,14 @@
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <meta name="version" content="{{ site.version }}">
 
+  <!-- Tealium Tag Manager - SYNC -->
+  {% if jekyll.environment == 'production' %}
```
[the remark this answers is not in the document]
Comment (kyeah) [reply _includes/_head.html:8]: Created ticket here: https://jira.cms.gov/browse/DPC-3650

**Review by kyeah — APPROVED** [review]

**Pull request merged**